src/osd/ReplicatedBackend.cc

```diff
@@ -1295,6 +1295,14 @@ void ReplicatedBackend::calc_head_subsets(
   uint64_t size = obc->obs.oi.size;
   if (size)
     data_subset.insert(0, size);
+  if (parent->min_peer_features() & CEPH_OSD_PARTIAL_RECOVERY) {
+    const auto it = missing.get_items().find(head);
+    assert(it != missing.get_items().end());
+    data_subset.intersection_of(it->second.clean_regions.get_dirty_regions());
+    dout(10) << "calc_head_subsets " << head
+             << "  data_subset " << data_subset << dendl;
+    return;
```

Comment:
I agree to the second part.  I don't understand what you mean by the first part.